### PR TITLE
[fern] TTA y-mirror averaging to symmetrize wsy predictions (Wave 20)

### PR DESCRIPTION
**Pull request by morganmcg1**
# [fern] Test-time augmentation: y-mirror + small rotation averaging (Wave 20)

## Hypothesis

**Inference-time augmentation can recover wsy/wsz error without retraining.** DrivAerML vehicles are approximately bilaterally symmetric; the y-mirror is a near-exact symmetry of the underlying physics:
- `surface_pressure(x, y, z) = surface_pressure(x, -y, z)` (scalar, unchanged)
- `volume_pressure(x, y, z) = volume_pressure(x, -y, z)` (scalar, unchanged)
- `wsx(x, y, z) = wsx(x, -y, z)` (vector x-component, unchanged)
- `wsy(x, y, z) = -wsy(x, -y, z)` (vector y-component, **sign flips**)
- `wsz(x, y, z) = wsz(x, -y, z)` (vector z-component, unchanged)

A trained model trained without mirror-augmentation breaks this symmetry by random initialization; averaging predictions from the original and y-mirrored input (with the wsy sign flip) produces a symmetrized estimate that should reduce wsy/wsz variance and bring val_abupt down at zero training cost.

This is the **free signal** noted in the research-state doc as direction #5 — and unlike the training-time mirror-aug experiment (tanjiro #443) which is currently plateaued at 8.0%, TTA does not require retraining; it just needs a clean inference path on the existing baseline checkpoint.

## Implementation

Add a `--tta-mirror` flag to `train.py` that activates only during validation/test:
1. Run forward pass on the original input → predictions P_orig.
2. Mirror the input geometry along the y-axis (negate y-coordinates of surface and volume points). Surface normals (if loaded) also need y-flipping.
3. Run forward pass on the mirrored input → predictions P_mirror.
4. Mirror P_mirror back: flip the sign of `wsy` channels in the mirrored predictions, leave `wsx`, `wsz`, surface_pressure, volume_pressure unchanged.
5. Final prediction = 0.5 × (P_orig + mirrored-back P_mirror).

This is implemented in the val loop only — no training-time changes, no extra params, no extra training compute. Doubles validation time (acceptable; val is a small fraction of training time).

**Optional second arm (Trial B):** add small-angle rotation TTA. For DrivAerML the geometry is roughly aligned to flow direction so large rotations are not a symmetry — but ±2° yaw rotation may capture local frame robustness. Skip for Trial A unless Trial A wins.

## Trial plan

**Trial A — y-mirror TTA from baseline checkpoint (zero training cost, val-only):**
- Load PR #174 best checkpoint (W&B run `vu4jsiic`, ep~45.3 best). Run a single validation pass with `--tta-mirror`. Report all val_primary metrics with and without TTA. **No 50-epoch training needed if a checkpoint-only val script exists**; if not, the simplest implementation is to add the TTA path and run a 1-epoch "training" with `--epochs 1 --validation-every 1` from the checkpoint, then read the val log.
- Decision rule: if val_abupt drops by ≥0.10pp vs the baseline ep45 number (6.9549% → 6.85% or lower), this is a merge.

**Trial B — Train from scratch with TTA enabled in val (sanity)**:
- Only if Trial A wins. Confirms the TTA gain is robust across checkpoints, not a single-seed fluke. Use the standard 5L/256d 50-epoch recipe.

## Reproduce command (Trial A — checkpoint val)

```bash
# First, ensure checkpoint loading + TTA val path is implemented.
# Then run:
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 --model-hidden-dim 256 --model-heads 4 \
  --fourier-pe --no-use-ema \
  --lr 3e-4 --lr-cosine-t-max 50 \
  --no-compile-model \
  --tta-mirror \
  --epochs 1 --validation-every 1 \
  --wandb-group bengio-wave20 --wandb-name fern/tta-mirror-trialA --agent fern
# (load the PR #174 vu4jsiic best checkpoint into the model before the val pass)
```

## Reproduce command (Trial B — full retrain with TTA val)

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 --model-hidden-dim 256 --model-heads 4 \
  --fourier-pe --no-use-ema \
  --lr 3e-4 --lr-cosine-t-max 50 \
  --no-compile-model \
  --tta-mirror \
  --kill-thresholds "500:train/loss<5,89000:val_primary/abupt_axis_mean_rel_l2_pct<13,178000:val_primary/abupt_axis_mean_rel_l2_pct<10.5" \
  --wandb-group bengio-wave20 --wandb-name fern/tta-mirror-trialB --agent fern
```

## Kill gates (Trial B only — Trial A is a single-pass val)
- EP5 (step 89,080) abupt > 13.0% → kill
- EP10 (step 178,160) abupt > 10.5% → kill
- EP15 (step 267,240) abupt > 8.5% → kill

## Baseline (target to beat)

| Metric | PR #174 (vu4jsiic, ep45.3) | TTA goal |
|--------|------------------------------|----------|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **6.9549** | < 6.85% |
| `val_primary/wall_shear_y_rel_l2_pct` | 8.7345 | < 8.0% (TTA primarily helps wsy via sign flip) |
| `val_primary/wall_shear_z_rel_l2_pct` | 10.5766 | < 10.3% (small-magnitude gain expected) |
| `val_primary/wall_shear_x_rel_l2_pct` | 7.109 | < 7.0% (small gain via averaging variance) |

## Why this hypothesis is high-leverage

- Pure inference-time signal — zero training cost, no GPU-time risk.
- Direct physical motivation: y-mirror is a near-exact symmetry of vehicle aerodynamics.
- Composable: if TTA wins, it stacks with every other improvement (including future tangent-frame, SW=2.0, mirror-aug-train, etc.).
- The sign-flip on wsy mathematically symmetrizes the prediction distribution along the binding axis.
- Should generalize to test set without further training.

## Constraints (do not break)

- `--no-compile-model` mandatory.
- `--fourier-pe` mandatory.
- `--no-use-ema` mandatory.
- TTA must be val-only (training loss path unchanged) — else the experiment is no longer a TTA test.
- One PR, one hypothesis. No bundling other changes.

### CODE REVIEW

**Comment by morganmcg1** [comment]
STUDENT fern: Your PR has been assigned. Please post your W&B run ID and confirm you have started training. The baseline to beat is **val_abupt = 6.9549%** (PR #174, run `vu4jsiic`). Post your ep1 metrics as soon as they are available and apply the standard kill gates (ep5: >13%, ep10: >10.5%, ep15: >8.5%, ep20: >8.0%).

**Comment by morganmcg1** [comment]
ADVISOR (infra-block escalation, 2026-05-03): Pod `senpai-bengio-fern` is confirmed blocked by a zombie `train.py` process from a prior wave. This is a known infrastructure issue (#466) — **you are not at fault**. The human infrastructure team has been notified and must SIGKILL the zombie process and restart the deployment.

Once the pod restarts and is free, launch your coord-norm + y-mirror TTA run immediately per the PR instructions. Post your W&B run ID(s) and ep1 metrics as soon as available. No further action needed from you until the pod is freed.
